### PR TITLE
refactor(wallet-mobile): Button, IconProps, useCopy

### PR DESCRIPTION
**Pull request by jorbuedo**
##### Ticket

[YOMO-1938](https://emurgo.atlassian.net/browse/YOMO-1938)

https://www.figma.com/design/cnvVqO3XxEPNFyE5qE1j2s/Components---Mobile?node-id=2065-18310&node-type=frame&t=OPYP3NRW5kwfsQgQ-0

Relevant bits:

- apps/wallet-mobile/src/components/Button/Button.tsx
- apps/wallet-mobile/src/components/Clipboard/ClipboardProvider.tsx
- apps/wallet-mobile/src/components/CopyButton.tsx

Will need QA review due to how many there are, but seem fine from what I've seen:

- Icons sizes and colors
- Buttons sizes, colors and positions
- Click to copy buttons and fields

[YOMO-1938]: https://emurgo.atlassian.net/browse/YOMO-1938?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/wallet-mobile/src/components/Icon/Ada.tsx [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import React from 'react'
 import Svg, {Path} from 'react-native-svg'
 
-type Props = {size: number; color?: string}
+import {IconProps} from '.'
```
Comment (michalszorad) [inline apps/wallet-mobile/src/components/Icon/Ada.tsx]: Will it create a circular dependency?

File: apps/wallet-mobile/src/components/Button/Button.tsx [hunk 2]
```diff
@@ -1,203 +1,195 @@
...
+  Primary: 'Primary',
+  Secondary: 'Secondary',
+  Critical: 'Critical',
+  Text: 'Text',
+  SecondaryText: 'SecondaryText',
+  Circle: 'Circle',
+  Link: 'Link',
+} as const
+
+export type ButtonType = (typeof ButtonType)[keyof typeof ButtonType]
```
Comment (michalszorad) [inline apps/wallet-mobile/src/components/Button/Button.tsx:18]: Isn't a string union better? `export type ButtonType = 'Primary' | 'Secondary' | ...`

File: apps/wallet-mobile/src/components/Icon/Ada.tsx (hunk 1 above)
Comment (jorbuedo) [reply apps/wallet-mobile/src/components/Icon/Ada.tsx]: Mmm I haven't noticed anything, is there any way to know?

File: apps/wallet-mobile/src/components/Button/Button.tsx (hunk 2 above)
Comment (jorbuedo) [reply apps/wallet-mobile/src/components/Button/Button.tsx:18]: It exports the same thing (a string union) without having to add new ones later.
Can be used as just a string as well, although I prefer the ButtonType reference so it's easier to find them later.

image

File: apps/wallet-mobile/src/components/Icon/Ada.tsx (hunk 1 above)
Comment (jorbuedo) [reply apps/wallet-mobile/src/components/Icon/Ada.tsx]: Changing it anyway

**Review by stackchain — APPROVED** [review]

**Comment by stackchain** [comment]
@jorbuedo can you tackle the conflicts?

**Pull request merged**